docs/lazy-adr/adr-006-row-propagation.md

```diff
@@ -1,56 +1,195 @@
+##### Proposal Stage
+Previously, the State in proposal stage waited for all Parts to assemble the entire Block. Instead, the State waits for
+the half of all Rows from a proposer and/or peers to recompute the Block's data and notifies them back that no more
+needs to be sent. Also, through Rows, only minimally required amount of information is gossiped. Everything else to
+assemble the full Block is collected from own chain State and Proposal.
 
 ## Status
 Proposed
 
 ## Consequences
 ### Positive
-* We don't have all the cons of "nah it works just don't touch it" approach
+* Hardening of consensus gossiping with erasure coding
+* Blocks exceeding the size limit are immediately rejected on Proposal, without the need to download an entire Block.
+* More control over Row message size during consensus, comparing to Part message, as last part of the block always has
+  unpredictable size. `DAHeader`, on the other hand, allows knowing precisely the size of Row messages.
+* Less bandwidth usage
+  * Only required Block's data is gossiped.
+  * Merkle proofs of Parts are not sent on the wire
+* Only one system-wide block data commitment schema
 * We don't abandon the work we were doing for months and taking profits out of it
-    * PR [#287](https://github.com/celestiaorg/lazyledger-core/pull/287)
-    * PR [#312](https://github.com/celestiaorg/lazyledger-core/pull/312)
-    * PR [#427](https://github.com/celestiaorg/lazyledger-core/pull/427)
-    * and merged others
+  * PR [#287](https://github.com/celestiaorg/lazyledger-core/pull/287)
+  * PR [#312](https://github.com/celestiaorg/lazyledger-core/pull/312)
+  * PR [#427](https://github.com/celestiaorg/lazyledger-core/pull/427)
+  * and merged others
 
 ### Negative
-* We invest some more time(~1.5 weeks). Although most of the work is done and we need to do the latest spurt to rewrite tests and merge all of the work.
+* We invest some more time(~1.5 weeks).
+  * Most of the work is done. Only few changes left in the implementation along with peer reviews.
 
 ### Neutral
-* Initially, we send redundant block data, but later we can fix that after rsmt2d API changes.
+* Rows vs Parts on the wire
+  * Previously, parts were propagated with max size of 64KiB. Let's now take a Row of the largest 128x128 block in
+    comparison. The actual data size in such a case for the Row would be 128x256(shares_per_row*share_size)=32KiB, which
+    is exactly two times smaller than a Part.
+* Gossiped chunks are no more constant size. Instead, their size is proportional to the size of Block's data.
+* Another step back from original Tendermint's codebase
```

Comment:
I'd say that any change we make to the tendermint (which aren't upstreamed) comes with more costs than just the implementation, e.g.:
- audits
- more difficult to keep the code base up to data with upstream 
- less battle-tested in practice

So I'd say this, in general, any change we make (especially in the consensus package) should be treated with the greatest care (and only do them in the first place if we really do not see a way around it).